NOTICE-binary

```diff
@@ -5866,3 +5866,42 @@ This project bundles the following dependencies under the Apache Software Licens
 
 Apache HttpClient
 Copyright 1999-2017 The Apache Software Foundation
+
+// ------------------------------------------------------------------
+// NOTICE file corresponding to the section 4d of The Apache License,
+// Version 2.0, in this case for Apache Flink
+// ------------------------------------------------------------------
+
+Apache Flink
+Copyright 2006-2019 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+flink-python
```

Comment:
we already have a flink-python module in flink-dist; so we now have ambiguous licensing declarations. @sunjincheng121

Reply:
Thanks for point out this @zentol!
In our ( @aljoscha and me ) plan, we want to remove the `flink-libraries/flink-python` later, currently, we add the classifier “java-binding”  for the JAR.  for the licensing I agree with you, we really have ambiguous licensing declarations here.  :( 
We carefully thought about whether to change the module name, but the name flink-python is too apt, other names are not suitable, and the long-term goal we will remove the `flink-libraries/flink-python`. So we still want to use the name` flink-python`. So in order to eliminate ambiguity, can we add a little explanation in licensing? Or do you have better suggestions?

Reply:
well you could just call it flink-api-python or something, that's doesn't seem that inappropriate.

I don't believe we should be re-using artifactIds between releases if they are fundamentally different artifacts; it is for the same reason that we have not added another `flink-ml` module but named it `flink-ml-parent` instead.

Reply:
Currently, the names of modules in flink is `flink-language` patterns, such as `flink-java`, `flink-scala`. for the goal of `flink-python` will add  `user-defined-functions` support in the next release,  for UDFs we will add `data-service`, `state-service`...etc.  And it's not only the API level. These functions need to be integrated with the beam. The code inside flink will not be very much, so the current plan is to distinguish these codes with packages or folder(for python). So, If we should not re-using artifactId, how about using `flink-py`. @zentol @aljoscha  

BTW: I appreciate if we can discuss this module issue in the [design doc](https://docs.google.com/document/d/1ybYt-0xWRMa1Yf5VsuqGRtOfJBz4p74ZmDxZYg3j_h8/edit?usp=sharing):  https://docs.google.com/document/d/1ybYt-0xWRMa1Yf5VsuqGRtOfJBz4p74ZmDxZYg3j_h8/edit?usp=sharing